Jenkinsfile

```diff
@@ -0,0 +1,149 @@
+        builds[bldtype] = {
+            node('rippled-dev') {
+                checkout scm
+                dir ('build') {
+                    deleteDir()
+                }
+                def cdir = upDir(pwd())
+                echo "BASEDIR: ${cdir}"
+                def compiler = getCompiler(bldtype)
+                def target = getTarget(bldtype)
+                if (compiler == "coverage") {
+                    compiler = 'gcc'
+                }
+                echo "COMPILER: ${compiler}"
+                echo "TARGET: ${target}"
+                def clang_cc  = (compiler == "clang") ? '/opt/llvm-4.0.0/bin/clang' : ''
+                def clang_cxx = (compiler == "clang") ? '/opt/llvm-4.0.0/bin/clang++' : ''
+                def ucc = isNoUnity(target) ? 'true' : 'false'
+                echo "USE_CC: ${ucc}"
+                withEnv(["CCACHE_BASEDIR=${cdir}",
+                         'LCOV_ROOT=""',
+                         "TARGET=${target}",
+                         "CC=${compiler}",
+                         'BUILD=cmake',
+                         'VERBOSE_BUILD=true',
+                         "CLANG_CC=${clang_cc}",
+                         "CLANG_CXX=${clang_cxx}",
+                         "CCACHE_LOGFILE=${bldtype}.ccache.txt",
+                         "USE_CCACHE=${ucc}"])
+                {
+                    myStage(bldtype)
+                    try {
+                        if (fileExists("${bldtype}.ccache.txt")) {
+                            sh "rm -f ${bldtype}.ccache.txt"
+                        }
+                        sh "ccache -s > ${bldtype}.txt"
+                        // the devtoolset from SCL gives us a recent gcc. It's not strictly needed
+                        // when we are building with clang, but it doesn't seem to interfere either
+                        sh "source /opt/rh/devtoolset-6/enable && (/usr/bin/time -p ./bin/ci/ubuntu/build-and-test.sh 2>&1) 2>&1 >> ${bldtype}.txt"
+                        sh "ccache -s >> ${bldtype}.txt"
+                    }
+                    catch (any) {
+                        throw any
+                    }
+                    finally {
+                        def outstr = readFile("${bldtype}.txt")
+                        def st = getResults(outstr)
+                        def time = getTime(outstr)
+                        def txtcolor = getFailures(outstr) == 0 ? "DarkGreen" : "Crimson"
+                        outstr = null
+                        def shortbld = bldtype
+                        shortbld = shortbld.replace('debug', 'dbg')
+                        shortbld = shortbld.replace('release', 'rel')
+                        manager.addShortText("${shortbld}: ${st}, t: ${time}", txtcolor, "white", "0px", "white")
+                        archive "${bldtype}.txt"
+                        if (ucc == "true" && fileExists("${bldtype}.ccache.txt")) {
+                            archive "${bldtype}.ccache.txt"
```

Comment:
we're only enabling `ccache` for nonunity builds currently (`ucc == true`). This archiving stuff was related to the `ccache` logfile, which isn't working anyhow. The intention is that the logfile could be used to figure out why ccache is not always getting hits, but the logfile is presently always empty. I think it's best to just remove this logfile for now since it provides no benefit. As to whether we should just enable ccache for all builds, I'm open to suggestions. The only risk I can see is if `cacche` has a bug in how it invokes the compiler, it might be nice to have some builds using it and some not. I'm open to suggestions.